pkg/generator/generator.go

```diff
@@ -82,10 +82,14 @@ func addIngressToCaches(caches *Caches,
 	if err != nil {
 		return err
 	}
-	if ingressTranslation != nil {
-		caches.AddTranslatedIngress(ingress, ingressTranslation)
-	}
 
+	if ingressTranslation == nil {
+		return nil
+	}
+	if err := caches.ValidateIngress(ingressTranslation); err != nil {
```

Comment:
I think it would be better to make `ValidateIngress` private and call it first thing in the `AddTranslatedIngress` function.
The reason is that as it is now, the user of the `caches` needs to remember to always call `ValidateIngress` before calling `AddTranslatedIngress`. This might cause bugs later if we need to call `AddTranslatedIngress` from a different part of the codebase.